### PR TITLE
Update Envoy to d417618 2024/04/15

### PR DESCRIPTION
**Pull request by qqustc**
- sync files
- add `# unique` in .bazelrc to avoid CI failure https://dev.azure.com/cncf/4684fb3d-0389-4e0b-8251-221942316e06/_apis/build/builds/168119/logs/57
- add `cluster_manager_->initialize(bootstrap_);` to fix a CI crash which is introduced by https://github.com/envoyproxy/envoy/pull/33221

### CODE REVIEW

**Comment by qqustc** [comment]
retest/

**Comment by qqustc** [comment]
retest\

File: .bazelrc [hunk 1]
```diff
@@ -141,11 +142,11 @@ build:asan --copt -O1
 build:asan --copt -fno-optimize-sibling-calls
 
 # Clang ASAN/UBSAN
-build:clang-asan --config=clang
-build:clang-asan --config=asan
-build:clang-asan --linkopt -fuse-ld=lld
-build:clang-asan --linkopt --rtlib=compiler-rt
-build:clang-asan --linkopt --unwindlib=libgcc
+build:clang-asan --config=clang                                                             # unique
```
Comment (mum4k) [inline .bazelrc:145]: As discussed offline, please wait for and pull https://github.com/envoyproxy/envoy/pull/33605 in. Let's keep this file as close to the Envoy's version as possible.

File: source/client/process_impl.cc [hunk 2]
```diff
@@ -902,6 +902,7 @@ bool ProcessImpl::runInternal(OutputCollector& collector, const UriPtr& tracing_
     }
     ENVOY_LOG(debug, "Computed configuration: {}", absl::StrCat(bootstrap_));
     cluster_manager_ = cluster_manager_factory_->clusterManagerFromProto(bootstrap_);
+    THROW_IF_NOT_OK(cluster_manager_->initialize(bootstrap_));
```
Comment (mum4k) [inline source/client/process_impl.cc:905]: Let's not throw here. The pattern in this function is to handle errors in the following way:

```cc
if (!status.ok()) {
      ENVOY_LOG(error, "error message");
      return false;
}
```

**Review by mum4k — CHANGES_REQUESTED** [review]

File: .bazelrc (hunk 1 above)
Comment (qqustc) [reply .bazelrc]: Thanks.

File: source/client/process_impl.cc (hunk 2 above)
Comment (qqustc) [reply source/client/process_impl.cc]: Done

**Review by mum4k — APPROVED** [review]

**Pull request merged**